### DISCUSSION TITLE
Differential vs RSE Measurement

### DISCUSSION THREAD

**Discussion by osamaabualsoud** [ni/nidaqmx-python#459] [Q&A]
Hello, my script seems to be returning analog input readings in differential mode (using an NI 6001). Is this a setting I make within my script? I couldn't find an option for this in the examples.

Also, sometimes a voltages will return at very unexpected values. If I supply 4 volts to ai0:4, four of the readings will read correctly and the last one may read 2.6v. Any recommendations for this? I tried other ports, to no avail. Possibly a faulty card, or a timing issue?

Thank you very much for your time

**Comment by zhindes** [comment]
Yes, you can change this in the add channel method, e.g. `add_ai_voltage_chan`. There is an optional parameter for `terminal_config`. Here is an example documentation: https://nidaqmx-python.readthedocs.io/en/latest/ai_channel_collection.html#nidaqmx._task_modules.ai_channel_collection.AIChannelCollection.add_ai_voltage_chan

**Reply by osamaabualsoud** [reply]
Hello, can you show an example of how to set it to RSE mode? I read through the docs but didn't see a way to adjust it. 

I tried this based on the info I found, but it will not compile: read_task.ai_channels.add_ai_voltage_chan("Dev1/ai0",terminal_config=10083)

**Reply by bkeryan** [reply]
Here's a test that sets RSE mode: https://github.com/ni/nidaqmx-python/blob/7f2849a231a4bdfc5b89d1f75fbe886c3117646a/tests/legacy/test_channel_creation.py#L71

You need to use the enum, not a bare integer.

```
from nidaqmx.constants import TerminalConfiguration
...
terminal_config=TerminalConfiguration.RSE
```
